patches/server/0977-Add-PlayerShieldDisableEvent.patch

```diff
@@ -0,0 +1,64 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cryptite <cryptite@gmail.com>
+Date: Mon, 1 May 2023 16:22:43 -0500
+Subject: [PATCH] Add PlayerShieldDisableEvent
+
+Called whenever a player's shield is disable. The cooldown (default 100 ticks) is also provided to optionally be changed
+
+diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
+index 02cb6b8c1d59855ff4a8aad3024fe12007eca0ee..0636403ac05cbaa0e6b55a2ffa96d7ca40641f19 100644
+--- a/src/main/java/net/minecraft/world/entity/Mob.java
++++ b/src/main/java/net/minecraft/world/entity/Mob.java
+@@ -1686,7 +1686,11 @@ public abstract class Mob extends LivingEntity implements Targeting {
+             float f = 0.25F + (float) EnchantmentHelper.getBlockEfficiency(this) * 0.05F;
+ 
+             if (this.random.nextFloat() < f) {
+-                player.getCooldowns().addCooldown(Items.SHIELD, 100);
++                // Paper start
++                io.papermc.paper.event.player.PlayerShieldDisableEvent shieldDisableEvent = new io.papermc.paper.event.player.PlayerShieldDisableEvent((org.bukkit.entity.Player) player.getBukkitEntity(), getBukkitEntity(), 100);
++                if (!shieldDisableEvent.callEvent()) return;
++                player.getCooldowns().addCooldown(Items.SHIELD, shieldDisableEvent.getCooldown());
```

Comment:
You need some way to bypass the `PlayerItemCooldownEvent` that is fired inside addCooldown because we just fired it.